diff -Naur onvif_srvd.ori/src/daemon.c onvif_srvd/src/daemon.c
--- onvif_srvd.ori/src/daemon.c	2020-04-16 10:35:42.652985671 +0200
+++ onvif_srvd/src/daemon.c	2020-04-16 10:28:48.060441054 +0200
@@ -207,11 +207,11 @@
         return -1; // Could not create on PID file
 
 
-    if( lockf(fd, F_TLOCK, 0) == -1 )
-    {
-        close(fd);
-        return -1; // Could not get lock on PID file
-    }
+//    if( lockf(fd, F_TLOCK, 0) == -1 )
+//    {
+//        close(fd);
+//        return -1; // Could not get lock on PID file
+//    }
 
 
     if( ftruncate(fd, 0) != 0 )
