diff -Naur gsoap-2.8.ori/gsoap/src/soapcpp2_lex.l gsoap-2.8/gsoap/src/soapcpp2_lex.l
--- gsoap-2.8.ori/gsoap/src/soapcpp2_lex.l	2019-09-16 20:41:21.000000000 +0200
+++ gsoap-2.8/gsoap/src/soapcpp2_lex.l	2023-02-19 10:38:05.364363573 +0100
@@ -50,7 +50,7 @@
 extern int isatty(int); /* silence warnings */
 
 #ifdef WITH_BISON
-YYSTYPE yylval;
+extern YYSTYPE yylval;
 #undef YY_NEED_STRLEN
 #define YY_INPUT(buf, result, max_size) \
         { \
