diff -Naur wsdd.ori/src/daemon.c wsdd/src/daemon.c
--- wsdd.ori/src/daemon.c	2020-06-22 14:30:19.000000000 +0200
+++ wsdd/src/daemon.c	2020-06-22 14:29:17.000000000 +0200
@@ -45,6 +45,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <errno.h>
+#include <fcntl.h>
 
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -206,11 +207,11 @@
         return -1; // Could not create on PID file
 
 
-    if( lockf(fd, F_TLOCK, 0) == -1 )
-    {
-        close(fd);
-        return -1; // Could not get lock on PID file
-    }
+//    if( lockf(fd, F_TLOCK, 0) == -1 )
+//    {
+//        close(fd);
+//        return -1; // Could not get lock on PID file
+//    }
 
 
     if( ftruncate(fd, 0) != 0 )
