Index: alsa-lib-1.1.4.1/src/conf/alsa.conf
===================================================================
--- alsa-lib-1.1.4.1.orig/src/conf/alsa.conf
+++ alsa-lib-1.1.4.1/src/conf/alsa.conf
@@ -77,7 +77,7 @@ defaults.pcm.ipc_gid audio
 defaults.pcm.ipc_perm 0660
 defaults.pcm.dmix.max_periods 0
 defaults.pcm.dmix.rate 48000
-defaults.pcm.dmix.format "unchanged"
+defaults.pcm.dmix.format "S16_LE"
 defaults.pcm.dmix.card defaults.pcm.card
 defaults.pcm.dmix.device defaults.pcm.device
 defaults.pcm.dsnoop.card defaults.pcm.card
